process/block/alteredaccounts/alteredAccountsProvider_test.go

```diff
@@ -358,7 +422,64 @@ func testExtractAlteredAccountsFromPoolShouldIncludeNFT(t *testing.T) {
 	require.NoError(t, err)
 
 	encodedAddr := hex.EncodeToString([]byte("addr"))
-	require.Equal(t, expectedToken.Value.String(), res[encodedAddr].Tokens[0].Balance)
+	require.Equal(t, &indexer.AccountTokenData{
+		Identifier: "token0",
+		Balance:    expectedToken.Value.String(),
+		Nonce:      expectedToken.TokenMetaData.Nonce,
+		MetaData:   expectedToken.TokenMetaData,
+	}, res[encodedAddr].Tokens[0])
+}
+
+func testExtractAlteredAccountsFromPoolShouldIncludeDestinationFromTokensLogsTopics(t *testing.T) {
+	t.Parallel()
+
+	receiverOnDestination := []byte("receiver on destination shard")
+	expectedToken := esdt.ESDigitalToken{
+		Value: big.NewInt(37),
+		TokenMetaData: &esdt.MetaData{
+			Nonce: 38,
+		},
+	}
+	args := getMockArgs()
+	args.AccountsDB = &state.AccountsStub{
+		LoadAccountCalled: func(_ []byte) (vmcommon.AccountHandler, error) {
+			return &state.UserAccountStub{
+				RetrieveValueFromDataTrieTrackerCalled: func(_ []byte) ([]byte, error) {
+					tokenBytes, _ := args.Marshalizer.Marshal(expectedToken)
+					return tokenBytes, nil
+				},
+			}, nil
+		},
+	}
+	aap, _ := NewAlteredAccountsProvider(args)
+
+	res, err := aap.ExtractAlteredAccountsFromPool(&indexer.Pool{
+		Logs: []*data.LogData{
+			{
+				LogHandler: &transaction.Log{
+					Address: []byte("addr"),
+					Events: []*transaction.Event{
+						{
+							Address:    []byte("addr"),
+							Identifier: []byte(core.BuiltInFunctionESDTNFTTransfer),
+							Topics: [][]byte{
+								[]byte("token0"),
+								big.NewInt(38).Bytes(),
+								nil,
+								receiverOnDestination,
```

Comment:
This test and the one above are identical, except that here you also add `receiver` address. You could: 
- Safely delete the one above, since it won't decrease any code coverage
- Also, below, instead of checking `len(res)`, please check the stored account token data for each address

Reply:
I was not really interested in the code coverage, but rather in testing as many scenarios as possible. This way, the 2 tests (even if they are >80% similar) will trigger different branches/flows that I wanted to check.

Reply:
Agree on the first point, but could you please follow-up on the second point?

Reply:
if you run the test `testExtractAlteredAccountsFromPoolShouldIncludeNFT` with coverage, you will notice that it won't enter within this `if`: `if len(topics) > idxReceiverAddressInTopics {` 
if you run the test `testExtractAlteredAccountsFromPoolShouldIncludeDestinationFromTokensLogsTopics` with coverage, you will notice that it will enter the same `if`